sgm/inference/api.py

```diff
@@ -242,17 +291,46 @@ def image_to_image(
             samples,
             force_uc_zero_embeddings=["txt"] if not self.specs.is_legacy else [],
             return_latents=return_latents,
-            filter=None,
+            filter=filter,
         )
 
+    def wrap_discretization(
+        self, discretization, image_strength=None, noise_strength=None, steps=None
+    ):
+        if isinstance(discretization, Img2ImgDiscretizationWrapper) or isinstance(
+            discretization, Txt2NoisyDiscretizationWrapper
+        ):
+            return discretization  # Already wrapped
+        if image_strength is not None and image_strength < 1.0 and image_strength > 0.0:
+            discretization = Img2ImgDiscretizationWrapper(
+                discretization, strength=image_strength
+            )
+
+        if (
+            noise_strength is not None
+            and noise_strength < 1.0
+            and noise_strength > 0.0
+            and steps is not None
+        ):

```

Comment:
Shouldn't this be before the previous if block? Or there should be a guard to prevent that noisy and image aren't used at the same time.